video-streaming-app-with-directus-and-sveltekit/index.md

```diff
@@ -0,0 +1,387 @@
+    alt={video.title}
+  />
+  <h3>{video.title}</h3>
+  <p>
+    {video.views} views • {new Date(video.upload_date).toLocaleDateString()}
+  </p>
+</a>
+
+<style>
+  .video-card {
+    display: block;
+    text-decoration: none;
+    color: inherit;
+  }
+  img {
+    width: 100%;
+    height: auto;
+  }
+</style>
+```
+Then add the following code to the `VideoGrid.svelte` file:
+
+```svelte
+<script lang="ts">
+  import type { Video } from "$lib/types";
+  import VideoCard from "./VideoCard.svelte";
+
+  export let videos: Video[];
+</script>
+
+<div class="video-grid">
+  {#each videos as video}
+    <VideoCard {video} />
+  {/each}
+</div>
+
+<style>
+  .video-grid {
+    display: grid;
+    grid-template-columns: repeat(auto-fill, minmax(250px, 1fr));
+    gap: 1rem;
+  }
+</style>
+```
+Those two files are reusable components created to render your video data and organize the videos in a grid layout.
+
+## Fetching video data
+In your `src/libs folder`, create a new folder named services. Inside this folder, create a file named `index.ts`. Add the following code to this file to use the Directus helper function to fetch all the videos from your Directus `videos` collection:
+
+```ts
+import getDirectusClient from "$lib/directus";
+import { readItems } from "@directus/sdk";
+import type { Video } from "$lib/types";
+
+export async function getVideos(params = {}): Promise<Video[]> {
+  const directus = getDirectusClient();
+  const response = await directus.request(readItems("videos", params));
+  return response as Video[];
+}
+```
+
+## Displaying thumbnails and titles
+Update the your `routes/+page.svelte` file to use the `getVideos` function to fetch video data and display it using the `VideoGrid` component. This will display the thumbnails, titles, views and dates of the videos.
+```svelte
++
+<script lang="ts">
+  import { onMount } from "svelte";
+  import { getVideos } from "$lib/services/index";
+  import VideoGrid from "$lib/components/VideoGrid.svelte";
+  import type { Video } from "$lib/types";
+
+  let videos: Video[] = [];
+
+  onMount(async () => {
+    try {
+      videos = await getVideos({
+        sort: ["-upload_date"],
+        limit: 20,
+        fields: ["*", "thumbnail.*", "video_file.*"],
+      });
+    } catch (error) {
+      console.error("Error fetching videos:", error);
+    }
+  });
+</script>
+
+<h1>Stream your favourite vidoes</h1>
+
+
+{#if videos.length > 0}
+  <VideoGrid {videos} />
+{:else}
+  <p>Loading videos...</p>
+{/if}
+```
+Directus stores file metadata in the `directus_files` collection. When using file fields in other collections, Directus creates a one-to-many relationship. To include file data when fetching items, you use dot notation in the fields parameter, like `thumbnail.*` and `video_file.*`. This tells Directus to populate the file information from the `directus_files` collection.
+
+![Video Listing](<Screenshot 2024-07-04 at 11.53.25.png>)
```

Comment:
````suggestion
## Displaying Thumbnails and Titles

Update the your `routes/+page.svelte` file to use the `getVideos` function to fetch video data and display it using the `VideoGrid` component. This will display the thumbnails, titles, views and dates of the videos.

```svelte
<script lang="ts">
  import { onMount } from "svelte";
  import { getVideos } from "$lib/services/index";
  import VideoGrid from "$lib/components/VideoGrid.svelte";
  import type { Video } from "$lib/types";

  let videos: Video[] = [];

  onMount(async () => {
    try {
      videos = await getVideos({
        sort: ["-upload_date"],
        limit: 20,
        fields: ["*", "thumbnail.*", "video_file.*"],
      });
    } catch (error) {
      console.error("Error fetching videos:", error);
    }
  });
</script>

<h1>Stream your favorite videos</h1>


{#if videos.length > 0}
  <VideoGrid {videos} />
{:else}
  <p>Loading videos...</p>
{/if}
```

Directus stores file metadata in the `directus_files` collection, and when using file fields in other collections, Directus creates a one-to-many relationship. To include file data when fetching items, you use dot notation in the fields parameter, like `thumbnail.*` and `video_file.*`. This tells Directus to populate the file information from the `directus_files` collection.

![Video Listing](<Screenshot 2024-07-04 at 11.53.25.png>)
````

Please update this to use the correct fields syntax. what you've done works but is not recommended. Nested items should be an object - refer to our docs for more.

Reply:
Done